test/integration-test.js

```diff
@@ -0,0 +1,73 @@
+let chai = require('chai');
+let PepperMint = require('../index')
+
+chai.should();
+
+try {
+  var config = require('./integration-config.json');
+} catch (e) {
+  // no config; don't run tests
+  process.exit()
+}
+
+describe('pepper-mint', function () {
+  describe('handles editing transactions', function () {
+    it('verifies fields have changed after editing', async function () {
+      this.timeout(30000);
+      let mint = await PepperMint(config.username, config.password, config.ius_session, config.thx_guid)
+      await createTransaction(mint)
+      let transactions = await getTransactions(mint)
+
+      await editTransaction(mint, transactions)
+      let newTnx = await getTransactions(mint)
+
+      await deleteTransaction(mint, newTnx)
+
+      await doAssertions(newTnx)
+    })
+  });
+});
+
+function doAssertions(transaction) {
+  transaction[0].merchant.should.equal("New Test Merchant Name")
+  transaction[0].category.should.equal("Vacation")
+  transaction[0].categoryId.should.equal(1504)
+  transaction[0].date.should.equal("May 5")
+  return transaction
+}
+
+function createTransaction(mint) {
+  let createRequest = {
+    amount: 1.23,
+    date: "05/04/2018",
+    merchant: "Test Merchant Name",
+    note: "This is a test transaction"
+  }
+  return mint.createTransaction(createRequest)
+}
+
+function getTransactions(mint) {
+  let getTransactionsRequest = {
+    query: [
+      "Test Merchant Name"
+    ],
+    startDate: new Date(2018, 4),
+    endDate: new Date(2018, 6)
+  }
+  return mint.getTransactions(getTransactionsRequest)
+}
+
+function editTransaction(mint, tnx) {
+  let editTransactionRequest = {
+    id: tnx[0].id,
+    merchant: "New Test Merchant Name",
+    category: "Vacation",
```

Comment:
It's been a while so I don't remember what all fields are accepted by this API, but for example if `notes` was set for the transaction, would this request delete that now that it's using `txnedit`?

Reply:
You're right, other fields such as notes, tags, etc. are allowed with this task. However, I was just hoping to validate changing this task would be fine on this PR and add a new schema for the args in another PR if that's alright with you. Just wanted to keep it small. Since your schema on the input args is restricted (people can't just add whatever fields they want) it should still work fine. Does that make sense?

Reply:
I see what you're saying. In your previous experience, any fields which you did not send were erased. I was actually worried about this as well but it does not seem to be the case. Let me see if I can write up a test case to prove this.

Reply:
I've implemented these tests.